sanic/websocket.py

```diff
@@ -7,23 +10,748 @@
+        """
+        Receive the next message frame by frame.
+        Return an iterator of :class:`str` for a text frame and :class:`bytes`
+        for a binary frame. The iterator should be exhausted, or else the
+        connection will become unusable.
+        With the exception of the return value, :meth:`recv_streaming` behaves
+        like :meth:`recv`.
+        """
+        if self.recv_lock.locked():
+            raise RuntimeError(
+                "cannot call recv_streaming while another task "
+                "is already waiting for the next message"
+            )
+        await self.recv_lock.acquire()
+        try:
+            self.can_pause = False
+            async for m in self.assembler.get_iter():
+                yield m
+        finally:
+            self.can_pause = True
+            self.recv_lock.release()
+
+    async def send(self, message: Union[Data, Iterable[Data]]) -> None:
+        """
+        Send a message.
+        A string (:class:`str`) is sent as a `Text frame`_. A bytestring or
+        bytes-like object (:class:`bytes`, :class:`bytearray`, or
+        :class:`memoryview`) is sent as a `Binary frame`_.
+        .. _Text frame: https://tools.ietf.org/html/rfc6455#section-5.6
+        .. _Binary frame: https://tools.ietf.org/html/rfc6455#section-5.6
+        :meth:`send` also accepts an iterable of strings, bytestrings, or
+        bytes-like objects. In that case the message is fragmented. Each item
+        is treated as a message fragment and sent in its own frame. All items
+        must be of the same type, or else :meth:`send` will raise a
+        :exc:`TypeError` and the connection will be closed.
+        :meth:`send` rejects dict-like objects because this is often an error.
+        If you wish to send the keys of a dict-like object as fragments, call
+        its :meth:`~dict.keys` method and pass the result to :meth:`send`.
+        :raises TypeError: for unsupported inputs
+        """
+        async with self.conn_mutex:
+
+            # TODO HANDLE THE SITUATION WHERE THE CONNECTION IS CLOSED
+
+            # Unfragmented message -- this case must be handled first because
+            # strings and bytes-like objects are iterable.
+
+            if isinstance(message, str):
+                self.connection.send_text(message.encode("utf-8"))
+                await self.send_data(self.connection.data_to_send())
+
+            elif isinstance(message, (bytes, bytearray, memoryview)):
+                self.connection.send_binary(message)
+                await self.send_data(self.connection.data_to_send())
+
+            # Catch a common mistake -- passing a dict to send().
+
+            elif isinstance(message, Mapping):
+                raise TypeError("data is a dict-like object")
+
+            # Fragmented message -- regular iterator.
+
+            elif isinstance(message, Iterable):
+                # TODO use an incremental encoder maybe?
```

Comment:
To be honest you don't really need to support fragmentation.

websockets had a bug where using compression + fragmentation crashed. Compression is enabled by default and it took year before someone reported it. So I'm pretty sure no one uses fragmentation.

You could ship without that feature and see if someone wants it!

Reply:
Ok. I'll remove that bit.